airflow/operators/python.py

```diff
@@ -232,7 +232,11 @@ def execute(self, context: Context) -> Any:
         context_merge(context, self.op_kwargs, templates_dict=self.templates_dict)
         self.op_kwargs = self.determine_kwargs(context)
 
-        return_value = self.execute_callable()
+        try:
+            dataset_events = context["dataset_events"]
+        except KeyError:
+            dataset_events = context["dataset_events"] = DatasetEventAccessors()
+        return_value = self.execute_callable(dataset_events)
```

Comment:
I am not sure why we need that change to add `dataset_events` to the calling args - we don't add other context information here as well. Why should be dataset event be different than other context vars?

Reply:
Because `execute_callable` does not receive the context at all. If you ask me it’s sort of designed quite weirdly to begin with.

I think I have a way to avoid not passing in the additional argument, by storing the value on the operator (`self`) instead. It’s a bit hacky and more of a workaround.